hypothesis-python/src/hypothesis/strategies/_internal/types.py

```diff
@@ -73,10 +73,30 @@
 except AttributeError:  # pragma: no cover
     pass  # `typing_extensions` might not be installed
 
+FinalTypes: tuple = ()
+try:
+    FinalTypes += (typing.Final,)  # type: ignore
+except AttributeError:  # pragma: no cover
+    pass  # Is missing for `python<3.8`
+try:
+    FinalTypes += (typing_extensions.Final,)
+except AttributeError:  # pragma: no cover
+    pass  # `typing_extensions` might not be installed
 
 # We use this variable to be sure that we are working with a type from `typing`:
 typing_root_type = (typing._Final, typing._GenericAlias)  # type: ignore
 
+# We use this to disallow all non-runtime types from being registered and resolved.
+# By "non-runtime" we mean: types that do not really exist in python's
+# and are just added for more fancy type annotations.
+# `Final` is a great example: it just indicates
+# that this value can't be reassigned.
+NON_RUNTIME_TYPES = {
+    "ClassVar": ClassVarTypes,
+    "TypeAlias": TypeAliasTypes,
+    "Final": FinalTypes,
+}
```

Comment:
I'm sorry to come back to this again, but... with the benefit of some space to think, it looks pretty redundant.  I'm also keen to avoid the loop - small-n loops are *strictly speaking* O(1), but still unaesthetic.

We could just put them in a single *set* (`if custom_type in NON_RUNTIME_TYPES:`) and get the name with `... instance of {custom_type .__name__}"`, and that's safe because we're not handling weird user-defined types which might not be named and hashable.